react-app/src/features/dashboard/index.tsx

```diff
@@ -40,7 +41,11 @@ export const Dashboard = () => {
   const { data: userObj } = useGetUser();
   const osData = useOsData();
 
-  if (userObj === undefined) {
+  if (
+    userObj === undefined ||
+    (userObj.user["custom:cms-roles"] &&
+      !Object.values(UserRoles).some((role) => userObj.user["custom:cms-roles"].includes(role)))
+  ) {
```

Comment:
Maybe extract this into a variable called `isAbleToAccessDashboard` or something to add context to the `if` check?